lib/kafka_ex/protocol/offset_commit.ex

```diff
@@ -11,7 +11,8 @@ defmodule KafkaEx.Protocol.OffsetCommit do
               topic: nil,
               partition: nil,
               offset: nil,
-              metadata: ""
+              metadata: "",
+              api_version: 0
 
     @type t :: %Request{
```

Comment:
Does this type need to be updated?

Reply:
Without it, the api_version can't be used consistently, i.e. [this line](https://github.com/kafkaex/kafka_ex/pull/342/files#diff-23354f817459dba990c9cab84bd1ab76R34) would need to check the type of the offset_commit_request

Reply:
(disclaimer: I might be understanding this incorrectly)

It seems to me that the API version should not be part of the request. Instead, we should determine what versions are supported by the server and use the highest.

See here for an example: https://github.com/kafkaex/kafka_ex/blob/master/lib/kafka_ex/server_0_p_10_and_later.ex#L184

My proposition means that:
- the version would be an additional argument of `KafkaEx.Protocol.OffsetCommit.create_request/3`
- in `server_0_p_8_*` and `server_0_p_9_*`, the version would be fixed
- in `server_0_p_10_and_later`, we would get the version from the server

Though this might be what's going to happen here and I just didn't understand correctly :).